### DISCUSSION TITLE
Slower generation speeds for every version after 1.68 -no solution-

### DISCUSSION THREAD

**Discussion by lightgauge** [YellowRoseCx/koboldcpp-rocm#70] [General]
For whatever reason, every version I've used has been taking at least twice as long after about 10 responses. I've used the same configuration file when trying out different versions. If I get to a point on 1.68 where my generation time is around 30 seconds, any version after that will take as long as 70+ seconds. Has anyone had this issue?

W10 22H2 
AMD 5600X 
AMD 6800XT 24.9.1

**Comment by Danik-droid** [comment]
I also have windows 10 and also Radeon only 6900 xt, I don't have such a problem. The last version ([KoboldCPP-v1.76.yr1-ROCm](https://github.com/YellowRoseCx/koboldcpp-rocm/releases/tag/v1.76.yr1-ROCm)) works great for me and is faster than vulkan in every way. Maybe try this version.

If that didn't work, I would first reinstall//update the AMD drivers. I would also test the system/card to see if there is any performance loss (maybe something is broken in the system or the hardware is failing) I would also check nocuda versions if I were you and see how it runs on vulkan. But theoretically our cards are not much different so the new rocm version should run much better.

Test it and let me know what you find.

**Reply by lightgauge** [reply]
I updated my drivers to 24.1.10 and restarted my pc. I'm testing a bit currently. I did a couple of runs on 1.68 with the responses coming in a little long at around 73 seconds while the 2 I did on 1.76 were 104 and 107. Vulkan took so long to generate I ended up closing it out as it was taking way too long. When I tried vulkan on the nocuda version, the first 2 runs were 46 and 43 seconds! I just did another and it came in at 78 seconds. That response was about as long as the ones I got on my 1.68 version so it's hard to tell if it's slightly slower or about the same. I'm just going off on regenerating responses. 

I'm undervolting the cpu and gpu, but I've already tried the gpu at stock with no change. I'll have to try everything stock and see. With my current settings, everything else runs fine. It's only when I got into AI stuff (SD generation shuts off my pc consistently) that I started having strange issues. I wonder if my 750w psu might be part of it when it comes to shutting off which actually happened once during these tests and I can't remember the last time that's happened or if it ever did when using kobold.

**Reply by Danik-droid** [reply]
It may be a hardware problem. I would do this:
- check the temperature (maybe it's too high and that's a GPU/CPU problem)
- reduce GPU clocking to e.g. 2000 and check how it works then.
- take a smaller model so as not to take up all the vram (maybe the vram memory chip is dying)
- check the power connections on the GPU (plugs - I had one loose in my old card and I had strange slowdowns on the GPU)
- generally I would restore the default settings in the BIOS (even below the standard settings and test slowly by increasing them.)
- if you have the possibility to test another PSU, I would do that.
- in the Radeon settings I would remove the shader cache (it shouldn't affect the rocm, but they sometimes like to break and cause problems.)
- apart from that I would probably disassemble everything and blow it with air and reconnect everything.

Good luck.

**Comment by lightgauge** [comment]
I'm at a total loss with this so I give up and will stick with 1.68. I just tried v1.78.yr0 and it's absolutely horrible. It's by far the worst version yet and makes my entire pc stutter. Something is definitely wrong with this as everything up to 1.68 runs fine and everything after has progressively gotten worse with each version. I appreciate the help.
